### PR TITLE
Add a new blog 0426

### PR DESCRIPTION
**Pull request by husiyu**
@amyangfei PTAL

### CODE REVIEW

File: dm-source-code-reading-4.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+### Process 实现细节
+
+该阶段的工作流程也很直观，通过 [一个收发数据类型为 `*pb.ProcessError` 的 `channel` 接收运行过程中出现的错误，出错后通过 context 的 `CancelFunc` 强制结束处理单元运行](https://github.com/pingcap/dm/blob/25f95ee08d008fb6469f0b172e432270aaa6be52/loader/loader.go#L408-L422)。在核心的 [数据导入函数](https://github.com/pingcap/dm/blob/25f95ee08d008fb6469f0b172e432270aaa6be52/loader/loader.go#L485) 中，工作模型与 mydumper 类似，即在 [主线程中分发任务](https://github.com/pingcap/dm/blob/25f95ee08d008fb6469f0b172e432270aaa6be52/loader/loader.go#L507)，[有多个工作线程执行具体的数据导入任务](https://github.com/pingcap/dm/blob/25f95ee08d008fb6469f0b172e432270aaa6be52/loader/loader.go#L500-L503)。具体的工作细节如下：
+
++ 主线程会按照库，表的顺序读取创建库语句文件 `<db-name>-schema-create.sql` 和建表语句文件 `<db-name>.<table-name>-schema-create.sql`，并在下游执行 SQL 创建相对应的库和表。
+
++ [主线程读取 `checkpoint` 信息，结合数据文件信息创建 fileJob 随机分发任务给一个工作子线程](https://github.com/pingcap/dm/blob/25f95ee08d008fb6469f0b172e432270aaa6be52/loader/loader.go#L944-L1015)，fileJob 任务的结构如下所示	：
+	
+	```
```
Comment (amyangfei) [inline dm-source-code-reading-4.md]:
```suggestion
	```go
```

File: dm-source-code-reading-4.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	   table         string
+	   dataFile   string
+	   offset       int64            // 表示读取文件的起始 offset，如果没有 checkpoint 断点信息该值为 0
+	   info           *tableInfo  // 保存原库表，目标库表，列名，insert 语句 column 名字列表等信息
+	}
+	```
+
++ 在每个工作线程内部，有一个循环不断从自己 `fileJobQueue` 获取任务，每次获取任务后会对文件进行解析，并将解析后的结果分批次打包为 SQL 语句分发给线程内部的另外一个工作协程，该工作协程负责处理 SQL 语句的执行。工作流程的伪代码如下所示，完整的代码参考 [`func (w *Worker) run()`](https://github.com/pingcap/dm/blob/25f95ee08d008fb6469f0b172e432270aaa6be52/loader/loader.go#L114-L173)：
+
+	```
```
Comment (amyangfei) [inline dm-source-code-reading-4.md]:
```suggestion
	```go
```

File: dm-source-code-reading-4.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	// worker 工作线程内分发给内部工作协程的任务结构
+	type dataJob struct {
```
Comment (amyangfei) [inline dm-source-code-reading-4.md]: It seems the variable fields are not alignment, I'm not sure whether we use space or tab here, it is better to make them align.

**Comment by husiyu** [comment]
@YiniXu9506 PTAL

**Comment by YiniXu9506** [comment]
LGTM

**Pull request merged**